docs/references/index.rst

```diff
@@ -29,6 +29,8 @@ Loaders
     datasets.fetch_vogue_cover_similarity
     datasets.fetch_things_similarity
     datasets.fetch_similarity_matrix
+    datasets.BaseManifold
```

Comment:
Please do not add BaseManifold here in the public docs. It is not interesting for 99% of the users.

Reply:
Will do and push in a few days